image_transport/include/image_transport/camera_publisher.h

```diff
@@ -74,7 +74,9 @@ class CameraPublisher
   CameraPublisher(
     rclcpp::Node * node,
     const std::string & base_topic,
-    rmw_qos_profile_t custom_qos = rmw_qos_profile_default);
+    const rclcpp::QoS & qos = rclcpp::QoS(10),
```

Comment:
Since in ROS 1 you needed to specify the `queue_size` and that's the aim of these changes (to get back to that behavior), I'd say do not default the `rclcpp::QoS` here.